src/rnp/fficli.cpp

```diff
@@ -1585,24 +1637,13 @@ cli_rnp_export_keys(rnp_cfg_t *cfg, cli_rnp_t *rnp, const char *filter)
         goto done;
     }
 
-    for (list_item *ki = list_front(keys); ki; ki = list_next(ki)) {
-        uint32_t         flags = base_flags;
-        rnp_key_handle_t key = *((rnp_key_handle_t *) ki);
-        bool             primary = false;
-        char *           grip = NULL;
+    for (auto key : keys) {
+        uint32_t flags = base_flags;
+        bool     primary = false;
 
         if (rnp_key_is_primary(key, &primary)) {
             goto done;
         }
-
-        /* skip subkeys which have primary key */
-        if (!primary && !rnp_key_get_primary_grip(key, &grip)) {
-            if (grip) {
-                rnp_buffer_destroy(grip);
-                continue;
-            }
-        }
-
```

Comment:
I understand that this logic is moved to the bottom of `key_matches_flags()` function. The problem is:
 it's not evident what `SUBKEYS_AFTER` means in the context of `matches`. `After` is the word that can be applied to `sorting` not `filtering`. It always takes me 30-60 minutes to figure out how this flag travels via functions as it is not documented. I suggest to remove this flag and retain this section as well as check the behaviour of `rnpkeys/print_keys_info()` as it is the only other place to use this `SUBKEYS_AFTER` flag.

Reply:
You are welcome to push additional comments which would help to understand things faster next time.